From ba1edd6097c7177fd457e2c1ad691b66930a6dc4 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 17 Jul 2024 08:55:18 +0200
Subject: [PATCH 1155/1695] drivers: scm-msg: fix clock rates description
 remaining count

Fix value of remaining clocks to describe on SCMI clock protocol
message CLOCK_DESCRIBE_RATES that does not take into account the
number of returned clock in the response.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 05d2d51bde8603c4c36bd34b9971b04203f86f1a drivers: scm-msg: fix clock rates description remaining count]
Change-Id: Ibaa1fd354acc1580cc59105605838a065aa9853e
Fixes: 90252e2a52c7 ("drivers: scmi-msg: clock adapts to output buffer size")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393703
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/scmi-msg/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/scmi-msg/clock.c b/core/drivers/scmi-msg/clock.c
index 0444b84c8..165cd2d9b 100644
--- a/core/drivers/scmi-msg/clock.c
+++ b/core/drivers/scmi-msg/clock.c
@@ -388,7 +388,7 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
 		}
 
 		out_count = rate_index - in_args->rate_index;
-		remaining = nb_rates - in_args->rate_index;
+		remaining = nb_rates - in_args->rate_index - out_count;
 		p2a.num_rates_flags = SCMI_RATES_BY_ARRAY(out_count, remaining);
 	} else if (status == SCMI_NOT_SUPPORTED) {
 		unsigned long triplet[3] = { 0, 0, 0 };
-- 
2.47.2

